tutorials/groups.qmd

```diff
@@ -0,0 +1,242 @@
+They can be represented in several ways, for example as angles of rotation (which corresponds to [`RealCircleGroup`](@ref)), unit complex numbers ([`CircleGroup`](@ref)), or rotation matrices ([`SpecialOrthogonal`](@ref)).
+Let's consider the last representation since it is the most nontrivial one and can be more easily generalized to other groups.
+
+The associated manifolds and groups are defined by:
+
+```{julia}
+G = SpecialOrthogonal(2)
+M = base_manifold(G)
+@assert M === Rotations(2)
+```
+
+This duality (Lie group and the underlying manifold being separate) is a common pattern in `Manifolds.jl`. The group `G` can be used for both Lie group-specific operations and metric-specific operation, while the manifold `M` only allows using manifold and metric operations. This way groups can be specialized in ways not relevant to plain manifolds, and if someone doesn't use the groups structure, they don't have to consider it by just using the manifold.
+
+Some basic definitions
+
+```{julia}
+# default basis
+B = DefaultOrthogonalBasis()
+# Identity rotation
+p0 = @SMatrix [1.0 0; 0 1]
+
+# Group identity element of a special type
+IG = Identity(G)
+```
+
+
+Let's say we want to define a manifold point `p_i` some rotation θ from the [`identity_element`](@ref) reference rotation `p0` (another point on the manifold that we will use as reference)
+
+```{julia}
+# + radians rotation from x-axis on plane to point i
+xθi = π/6
+```
+
+### From Coordinates
+
+To get our first Lie algebra element we can use the textbook [`hat`](@ref), or equivaliently a more generalized [`get_vector`](@ref), function:
+
+```{julia}
+X_ = hat(G, IG, xθi)              # specific definition to Lie groups
+xXi = get_vector(G, p0, xθi, B)   # generalized definition beyond Lie groups
+println(xXi)
+@assert isapprox(X_, xXi)
+```
+
+Note that `hat` here assumes a default (orthogonal) basis for the more general `get_vector`.
+
+::: {.callout-note}
+In this case, the same would work given the base manifold [`Rotations(2)`](@ref):
+```{julia}
+_X_ = hat(M, p0, xθi)             # Lie groups definition
+_X = get_vector(M, p0, xθi, B)   # generalized definition
+@assert _X_ == xXi; @assert _X == xXi
+```
+One more caveat here is that for the Rotation matrices, the tangent vectors are always stored as elements from the Lie algebra.
+:::
+
+Now, let's place this algebra element on the manifold using the exponential map [`exp`](@ref):
+```{julia}
+xRi = exp(G, p0, xXi)
+# similarly for known underlying manifold
+xRi_ = exp(M, p0, xXi)
+
+@assert isapprox(xRi, xRi_)
+```
+
+### To Coordinates
+
+The logarithm transform from the group back to algebra (or coordinates) is:
+```{julia}
+xXi_ = log(G, p0, xRi)
+xXi__ = log(M, p0, xRi)
+@assert xXi ≈ xXi__
+```
+
+Similarly, the coordinate value can be extracted from the algebra using [`vee`](@ref), or directly from the group using the more generalized [`get_coordinates`](@ref):
```

Comment:
“the coordinate value” is not correct I think? especially also because the function has a plural ;) but I think it is a bit vague, and one might explain, that the basis is just one basis vector here?

Reply:
I have rephrased this.